src/run_info.h

```diff
@@ -103,6 +103,13 @@ int RAI_InitRunInfo(RedisAI_RunInfo **result);
 
 int RAI_ShallowCopyDagRunInfo(RedisAI_RunInfo **result, RedisAI_RunInfo *src);
 
+/**
+ * Frees the shallow copy of RedisAI_RunInfo pointed by rinfo.
+ * @param rinfo copy to be freed.
+ * @retval The ref_count of the rinfo object after freeing this copy.
+ */
+long long RAI_DagRunInfoDecreaseFetch(RedisAI_RunInfo *rinfo);
```

Comment:
```suggestion
long long RAI_DagRunInfoFreeShallowCopy(RedisAI_RunInfo *rinfo);
```

Also, if it is not a big refactor, make something like
`RedisAI_RunInfo * RAI_DagRunInfoCreateShallowCopy(RedisAI_RunInfo *rinfo);`
we should avoid breaking the abstraction of the object.

Reply:
There is RAI_ShallowCopyDagRunInfo, but the ref_count increase is done outside the function (after we create n copies, we increase the ref_count by n). Should we move it to be inside the function?

Reply:
I think it's a good idea, it would have an equivalent effect